cadctap/cadctap/youcat.py

```diff
@@ -0,0 +1,290 @@
+        else:
+            m = magic.Magic()
+            t = m.from_file(table_defintion)
+            if 'XML' in t:
+                file_type = 'VOTable'
+            elif 'FITS' in t:
+                file_type = 'FITSTable'
+            elif 'ASCII' in t:
+                file_type = 'VOSITable'
+            else:
+                raise AttributeError('Cannot determine the type of the table '
+                                     'definition file {}'.
+                                     format(table_defintion))
+            logger.info('Assuming the table defintion file type: {}'.
+                        format(file_type))
+        logger.debug('Creating {} from file {} of type {}'.
+                     format(table_name, table_defintion, file_type))
+        headers = {'Content-Type': ALLOWED_TB_DEF_TYPES[file_type]}
+        self._tap_client.put((TABLES_CAPABILITY_ID, table_name),
+                             headers=headers,
+                             data=open(table_defintion, 'rb').read())
+        logger.debug('Successfully created table {}'.format(table_name))
+
+    def delete_table(self, table_name):
+        """
+        Deletes a table in youcat
+        :param table_name: name of the table to delete
+        """
+        if not table_name:
+            raise AttributeError('table name required in delete')
+
+        logger.debug('Deleting {}'.format(table_name))
+        self._tap_client.delete((TABLES_CAPABILITY_ID, table_name))
+        logger.debug('Successfully deleted table {}'.format(table_name))
+
+    def create_index(self, table_name, column_name, unique=False):
+        """
+        Creates a table index in youcat
+        :param table_name: name of the table
+        :param column_name: name of the column
+        :param unique: True if index is unique, False otherwise
+        """
+        if not table_name or not column_name:
+            raise AttributeError(
+                'table name and column required in index: {}/{}'.
+                format(table_name, column_name))
+
+        logger.debug('Index for column{} in table {}'.format(column_name,
+                                                             table_name))
+        result = self._tap_client.post((TABLE_UPDATE_CAPABILITY_ID, None),
+                                       data={'table': table_name,
+                                             'index': column_name,
+                                             'uniquer': unique},
+                                       allow_redirects=False)
+        if result.status_code == 303:
+            job_url = result.headers['Location']
+            if not job_url:
+                raise RuntimeError(
+                    'table update job location missing in response')
+            # start the job
+            self._tap_client.post('{}/phase'.format(job_url),
+                                  data={'PHASE': 'RUN'})
+            # start polling
+            short_waits = 10
```

Comment:
Doesn't look like short_waits is used...

Reply:
I think wait=1 while short_waits > 0 then when short_waits=0, wait=30 so when it does the tap_client.get it waits for however long wait is.